packages/sdk/src/instant-trade/InstantTradeBuilder.ts

```diff
@@ -40,11 +50,20 @@ export default class InstantTradeBuilder extends PSBTBuilder {
 
   setPrice(value: number) {
     this.validatePrice(value)
-    this.price = parseInt(value.toString())
+    this.price = parseInt(value.toString()) // intentional re-parsing to number as value can be floating point
   }
 
-  setRoyalty(value: number) {
-    this.royalty = value
+  setRoyalty(data: Omit<RoyaltyAttributes, "percentage">) {
+    if (data.amount < MINIMUM_AMOUNT_IN_SATS) return
```

Comment:
instead of `return` maybe we can throw a specific error instead?

Reply:
It's possible that the calculated royalty is less than the dust amount. In that case, we are supposed to drop the royalty payment. Throwing an error would halt the trade.